gradle/dependencies.gradle

```diff
@@ -47,23 +47,14 @@ if ( !versions.scala.contains('-') ) {
 }
 
 // mockito >= 5.5 is required for Java 21 and mockito 5.x requires at least Java 11
-// mockito 4.11 is used with Java 8 and Scala 2.13
-String mockitoVersion
-if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_11))
-  mockitoVersion = "5.10.0"
-else
-  mockitoVersion = "4.11.0"
+String mockitoVersion = "5.14.1"
```

Comment:
in my local machine, it has error about 
```
Caused by: java.lang.IllegalArgumentException: Java 23 (67) is not supported by the current version of Byte Buddy which officially supports Java 22 (66) - update Byte Buddy or set net.bytebuddy.experimental as a VM property
```
And I find mockito is resolve this problem in this [issue](https://github.com/mockito/mockito/pull/3459), thus I upgrade the mockito version.